### DISCUSSION TITLE
dynamic_proj needs sdm.mod models for abundance projections?

### DISCUSSION THREAD

**Discussion by rogerio-bio** [r-a-dobson/dynamicSDM#8] [Q&A]
Hello Dobson, 

It's me again 😅 

I managed to generate an abundance model with my data, but when using the dynamic_proj function, I get this error:

```
> dynamic_proj(dates = projection_dates,
+              projection.method = "abundance",
+              local.directory = "E:/abundance_teste/projection_covariates",
+              user.email = user.email,
+              sdm.mod = NULL,
+              sdm.weight = NULL,
+              sam.mod =  tapir_SAM_models,
+              sam.weight =  tapir_sam_weights,
+              cov.file.type = "csv",
+              save.directory = workingdirectory)
No sdm.thresh. Default 0.5
Error in UseMethod("predict") : 
  no applicable method for 'predict' applied to an object of class "NULL"
```

If I didn't understand wrong, in the help section of the function sdm.mod and sdm.weight are described as optional, but to generate the static abundance map I will need sdm.mod, is that it?

Thank you for the patience 😄

**Comment by r-a-dobson** [comment]
Hi Rogerio, 
To fix this error, remove `sdm.mod = NULL` and `sdm.weight = NULL`. All you have to do is not provide the arguments. If you still have issues with this then please email me (eerdo@leeds.ac.uk) with reproducible code for the error.
Rachel

**Reply by rogerio-bio** [reply]
Yep, i already tried that, but i get this message:

```
> dynamic_proj(dates = projection_dates,
+              projection.method = "abundance",
+              local.directory = "E:/abundance_teste/projection_covariates",
+              user.email = user.email,
+              sam.mod =  tapir_SAM_models,
+              sam.weight =  tapir_sam_weights,
+              cov.file.type = "csv",
+              save.directory = workingdirectory)
Error in dynamic_proj(dates = projection_dates, projection.method = "abundance",  : 
  argument "sdm.mod" is missing, with no default
```

**Reply by r-a-dobson** [reply]
Hi Rogerio,
Thank you for finding this bug. I have updated the function with the fix on GitHub. If you install the package again through `devtools::install_github("r-a-dobson/dynamicSDM")` then your code should work now.

I will send the updated package through to CRAN and it will be uploaded in the next couple of days. Let me know if you have any further issues with this. 

Rachel

**Reply by rogerio-bio** [reply]
Thank you 👍
